doc/reference/networking/lwm2m.rst

```diff
@@ -11,8 +11,281 @@ Overview
 ********
 
 Lightweight Machine to Machine (LwM2M) is an application layer protocol
-based on CoAP/UDP, and is designed to expose various resources for reading,
-writing and executing via an LwM2M server in a very lightweight environment.
+designed with device management, data reporting and device actuation in mind.
+Based on CoAP/UDP, LwM2M is a standard defined by the Open Mobile Alliance.
+Made suitable for constrained devices by using CoAP packet size optimization
+as well as it's simple, stateless flow which supports a ReST API.
+
+One of the key differences between LwM2M and CoAP is that an LwM2M client
+initiates the connection to an LwM2M server.  The server can then use the
+ReST API to manage various interfaces with the client.
+
+LwM2M uses a simple resource model with the core set of objects and resources
+defined in the specification.
+
+Example LwM2M object and resources: Device
+******************************************
+
+*Object definition*
+
++-----------+--------+----------+-----------+
+| Object ID | Name   | Instance | Mandatory |
++===========+========+==========+===========+
+|     3     | Device | Single   | Mandatory |
++-----------+--------+----------+-----------+
+
+*Resource definitions*
+
+``* R=Read, W=Write, E=Execute``
+
++----+-------------------+-----+----------+-----------+---------+
```

Comment:
This kind of table layout is rather hard to maintain.  While OK as is, you can also use a "list table" directive that's much easier to deal with, for example:
```
.. list-table:: Optional Table Title
   :header-rows: 1

   * - ID
     - Name
     - OP\*
     - Instance
     - Mandatory
     - Type

   * - 0
     - Manufacturer
     - R
     - Single
     - Optional
     - String
```
etc.

Reply:
Ah.. it was the first example in the list on the docs, so I used it.  Maybe we should reverse the order in the documentation's documentation?

Reply:
The "drawing" table layout is the only way to get row or column spans (where a cell spans more than one row or column), so it's the most functional.  But it's a mess to maintain when you need to edit the cell contents.  (There are some helpers in some text editors such as emacs.) The other table layouts (CSV and List) only let you have non-spanning cells -- you can leave cells blank, but the lines will still be there.  These latter two layouts are much easier to edit.  I'll take another look at the docs on this...